<launch>
  <!-- Arguments -->
  <arg name="robot_name" value="cube_stack_arm"/>
  <arg name="model" default="$(find cube_stack_env)/urdf/cube_stack_arm.xacro"/>

  <!-- Set essential parameters -->
  <param name="/cube_stack_arm/dynamixel_ids" value="1,2,3,4,5,"/>
  <param name="/cube_stack_arm/robot_description"
        command="xacro --inorder $(arg model)"/>

  <!--Setup TF -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" ns="/cube_stack_arm">
    <param name="publish_frequency" type="double" value="30.0" />
    <remap from="/joint_states" to="/cube_stack_arm/joint_states"/>
    <remap from="/robot_description" to="/cube_stack_arm/robot_description" />
  </node>
  <node type="static_transform_publisher" name="static_transform_publisher" pkg="tf" args="0.07 0 0.6 0 1.53 3.14 world camera_link 100" />

  <!-- Launch Arm custom commander -->
  <!-- <node pkg="cube_stack_env" type="roboarm_custom_commander" name="roboarm_custom_commander"/> -->

  <!-- Launch Arm ros_control -->
  <node ns="/cube_stack_arm" pkg="cube_stack_env" type="roboarm_hardware_interface" name="roboarm_hardware_interface"/>
  <rosparam command="load"
            file="$(find cube_stack_env)/config/cube_stack_arm_control.yaml"/>
  <node name="cube_stack_arm_controller_spawner" pkg="controller_manager" type="spawner"
    ns="/cube_stack_arm" args="joint_state_controller arm_trajectory_position_controller gripper_trajectory_position_controller"/>

</launch>
